.azure-devops/templates/run-tests-parallel.yml

```diff
@@ -93,6 +85,33 @@ steps:
           inlineScript: |
             export COVERAGE_FILE=.coverage.${{ parameters.name }}
             pytest -vv ${{ parameters.path }} -k "_int.py" --dist=loadfile -n ${{ parameters.num_threads }} --reruns ${{ parameters.num_reruns }} --reruns-delay ${{ parameters.reruns_delay }} --cov=azext_iot --cov-config .coveragerc --junitxml=junit/test-iotext-int.xml --deselect=azext_iot/tests/iothub/core/test_iothub_discovery_int.py::TestIoTHubDiscovery::test_iothub_targets --durations=0
+        ${{ if contains(parameters.path, 'azext_iot/tests/dps') }}:
+          inlineScript: |
+            IOTHUB=`echo test-dps-hub-py$RANDOM`
+            sudo apt install dos2unix -y -q
+            dos2unix pytest.ini
+            CREATE_HUB=`grep -c "azext_iot_testdps_hub=$" pytest.ini`
+
+            if (($CREATE_HUB > 0))
```

Comment:
Instead of relying on a failure, it is better to be explicit here, that we don't run this part of the script when **Azure CLI Version** is MSI. A failure should actually fail the pipeline